src/aposteriori/data_prep/create_frame_data_set.py

```diff
@@ -1109,39 +1184,39 @@ def _select_pdb_chain(
     output_pdb_path: pathlib.Path
         Output path with chain
     """
-    pdb_structure = ampal.load_pdb(pdb_path)
-    # Check if PDB structure is container and select assembly
-    if isinstance(pdb_structure, ampal.AmpalContainer):
+    # Check if PDB structure is container and select assembly:
+    chain_pdb = pdb_structure[chain]
+    if verbosity > 1:
         warnings.warn(
-            f"Selecting the first state from the NMR structure {pdb_structure.id}"
+            f"ATTENTION: You selected chain {chain}, for PDB code {pdb_name}. We will replace the original PDB file with the selected chain. Remove the 5th letter of your PDB code if this is unwanted behaviour."
         )
-        pdb_structure = pdb_structure[0]
-
-    chain_pdb = pdb_structure[chain]
-    warnings.warn(
-        f"ATTENTION: You selected chain {chain}, for PDB code {pdb_path}. We will replace the original PDB file with the selected chain. Remove the 5th letter of your PDB code if this is unwanted behaviour."
-    )
     # Save chain to file:
-    output_pdb_path = pathlib.Path(
-        str(pdb_path.with_suffix("")) + chain + pdb_path.suffix
-    )
-    with open(output_pdb_path, "w") as f:
+    pdb_name += chain
+    if nmr_state:
+        pdb_name += f"_{nmr_state}"
+    # Extract pdb extension:
+    ext_pdb = output_pdb_path.suffix
+    chain_pdb_path = output_pdb_path.parent / f"{pdb_name}{ext_pdb}"
+    # Save chain file
+    with open(chain_pdb_path, "w") as f:
         f.write(chain_pdb.pdb)
     # Delete original file
-    if pdb_path.exists():
-        pdb_path.unlink()
+    if output_pdb_path.exists():
+        output_pdb_path.unlink()
```

Comment:
Should we raise an error here and say that the file exists before deleting?

Reply:
An error or a warning? We have a warning earlier at line https://github.com/wells-wood-research/aposteriori/blob/3a1ea4ce8ec157276796ead0431a764d749030ff/src/aposteriori/data_prep/create_frame_data_set.py#L1190

Reply:
An error? Might be a bit dangerous to delete files if they're present, in case they gave the wrong name. I'd raise an error or change the name and raise a warning.

Reply:
the files are replaced only if they are downloaded. This is the only way we can deal with chain selection. Also in this case we are deleting the original pdb file with lots of nmr structures and creating one file for each state of nmr. We have to delete it for aposteriori to voxelise the whole folder.